pints/_log_pdfs.py

```diff
@@ -115,6 +115,161 @@ def sample(self, n=1):
+        self._pooled = np.asarray(pooled)
+        if len(self._pooled) != n_parameters:
+            raise ValueError(
+                'The array-like input `pooled` needs to have the same length '
+                'as the number of parameters of the individual log-pdfs.')
+
+        # Check that pooled contains only booleans
+        if self._pooled.dtype != np.bool:
+            raise ValueError(
+                'The array-like input `pooled` passed to PooledLogPDFs '
+                'has to contain booleans exclusively.')
+
+        # Get dimension of search space
+        self._n_pooled = np.sum(self._pooled)
+        n_individuals = len(self._log_pdfs)
+        self._n_unpooled = np.sum(~self._pooled)
+        self._n_parameters = \
+            self._n_pooled + n_individuals * self._n_unpooled
+
+    def __call__(self, parameters):
+        # Get parameters of pooled log-pdf
+        parameters = np.asarray(parameters)
+
+        # Create container for parameters of individuals log-pdf and fill with
+        # pooled parameters
+        params_ind = np.empty(shape=self._n_unpooled + self._n_pooled)
+        if self._n_pooled > 0:
+            params_ind[self._pooled] = parameters[
+                self._n_parameters - self._n_pooled:self._n_parameters]
+
+        # Compute pdf score
+        total = 0
+        for idx, pdf in enumerate(self._log_pdfs):
+            # Get unpooled parameters for individual
+            params_ind[~self._pooled] = parameters[
+                idx * self._n_unpooled: (idx + 1) * self._n_unpooled]
+
+            # Compute pdf score contribution
+            total += pdf(params_ind)
+        return total
+
+    def evaluateS1(self, parameters):
+        """
+        See :meth:`LogPDF.evaluateS1()`.
+
+        *This method only works if all the underlying :class:`LogPDF` objects
+        implement the optional method :meth:`LogPDF.evaluateS1()`!*
+        """
+        # Get parameters of pooled log-pdf
+        parameters = np.asarray(parameters)
+
+        # Create container for parameters of individuals log-pdf and fill with
+        # pooled parameters
+        params_ind = np.empty(shape=self._n_unpooled + self._n_pooled)
+        if self._n_pooled > 0:
+            params_ind[self._pooled] = parameters[
+                self._n_parameters - self._n_pooled:self._n_parameters]
+
+        # Compute pdf score and partials
+        total = 0
+        dtotal = np.zeros(shape=self._n_parameters)
+        for idx, pdf in enumerate(self._log_pdfs):
+            # Get unpooled parameters for individual
+            params_ind[~self._pooled] = parameters[
+                idx * self._n_unpooled: (idx + 1) * self._n_unpooled]
+
+            # Compute pdf score and partials for individual
+            score, partials = pdf.evaluateS1(params_ind)
+
+            # Add contributions to score and partials. Note that partials
+            # w.r.t. unpooled parameters receive only one contribution
```

Comment:
Can you explain or give ref to what you did here?

Reply:
I am not sure whether there is a ref. But I tried to extend the comment a little bit so it's hopefully clearer.

Essentially what is happening is that if you have a sum of likelihoods that have all independent parameters, then the partial with respect to any of the sum of likelihoods is equal to the partial of the associated individual likelihood. But if you pool a parameter across all likelihoods in the sum, a partial derivative with respect to that parameter equals to the sum of partials.

Reply:
Yeah, probably hard to find ref. Yes, I understand what you did, but maybe explain that in the docstring (not for me, for the users)? I don't know, @MichaelClerx what do you think?

Reply:
I thought it was more complicated than it is until I remembered the function is scalar.

So you just get `f=fi` and `df/dpj = dfi/dpj` for an unpooled parameter `pj` from model `i`
And for pooled ones it becomes `f=sum(fi)` and `df/dpj = sum(dfi/dpj)`

Reply:
I am happy to add something to the docstring though.

Reply:
Yes, be good to have this in the docstring, I think.

Reply:
Also, say or even emphasise that you've assumed the parameters are independent variables! Because the maths break if they are not, so it's important to mention that in the docstring too (if you haven't already done so).